### DISCUSSION TITLE
How can I extract messages from segment files?

### DISCUSSION THREAD

**Discussion by sFrenkie** [rabbitmq/rabbitmq-server#9558] [Questions]
We restarted rabbitmq after failed node. And we find 10GB of segment files under mnesia/quorum.
Is there a way how can I extract messages from this files?  It's binary - looks like erlang and messages are hardcoded.
Is there a way how to replay segment files back to rabbitmq?

**Comment by kjnilsson** [comment]
If you have some erlang skills you can use the Ra library ra_log_segment
module directly to read the individual segments:

https://github.com/rabbitmq/ra/blob/main/src/ra_log_segment.erl


that said, how did you end up in a situation where the queue isn't
functioning?

On Wed, 27 Sept 2023 at 16:13, sFrenkie ***@***.***> wrote:

> We restarted rabbitmq after failed node. And we find 10GB of segment files
> under mnesia/quorum.
> Is there a way how can I extract messages from this files? It's binary -
> looks like erlang and messages are hardcoded.
> Is there a way how to replay segment files back to rabbitmq?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rabbitmq/rabbitmq-server/discussions/9558>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJAHFCAZHPBZ3XFNZS6MTDX4Q7BLANCNFSM6AAAAAA5JQ6NRE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
*Karl Nilsson*
